### ISSUE TITLE
Continuation of #2094: unstable formatting

### ISSUE THREAD

**Issue by felixhekhorn** [psf/black#2137]
Hey @ichard26 !
Thanks a lot for your work! If I understood you correctly it is correct to open a new issue, right? (instead of commenting on [the old](https://github.com/psf/black/issues/2094)?)

are you sure, this is fixed? because it [still occurs](https://black.vercel.app/?version=master&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AQpAV5dAD2IimZxl1N_Wg094ywRrUqyV1Mc02BMgDEjlfXeIX8gJ44fDLD45kk18loYwAeOZlBBoeRUHrdsjt2rl5_lsbILjM7uXmCfBGa-UmtMuNZiyS1SPUNDCBPJQUvhw4wtC_mQQp-y99i1QqDGlauDNmGET3vwEswjCx1-XaTCYopYXa2f5JPyZ6fnu4DB-NUYeYAPHn92ijQhd25z0F-NWH8fSBwVSfEDrGIeRG-tQwVGigW5exiQpHuvp5jcLFatHupEgWdPLbGKdST6zLmvlON8hhv3_KxgcvgBF1lp1Z3AWwaIcg4olQbk-4wgJQo2r_fTxubDo4aWgo1qt4v-8yQGBLQUbWkCi2ZZNrfIxI0EWiOojDzZo6p3lSJW__vvVTPmet4R-lZxC9WxaOjoVKClgWE2ZQ0irefpm04YEKzTpJJpqWVAtcNiYAA23eSv6WLtt44f405Bf6-RJ90kAAAARwPl1khtcjgAAfoCqggAABsVWSGxxGf7AgAAAAAEWVo=) on master which includes PR #2126 by now

I'm not sure it is related to magic commata but instead I believe it is related to the comments: look at the diff:
```
assert sorted(
-        [*triv_ops, "T15.c+", "V15.c-"] #, "c+.c+"", "c-.c-"]
+        [*triv_ops, "T15.c+", "V15.c-"]  # , "c+.c+"", "c-.c-"]
     ) == get_ad_to_evol_map(3, [4])
```

and using this snippet (with one comment removed) does indeed [work as expected](https://black.vercel.app/?version=master&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AP5AVhdAD2IimZxl1N_Wg094ywRrUqyV1Mc02BMgDEjlfXeIX8gJ44fDLD45kk18loYwAeOZlBBoeRUHrdsjt2rl5_lsbILjM7uXmCfBGa-UmtMuNZiyS1SPUNDCBPJQUvhw4wtC_mQQp-y99i1QqDGlauDNmGET3vwEswjCx1-XaTCYopYXa2f5JPyZ6fnu4DB-NUYeYAPHn92ijQhd25z0F-NWH8fSBwVSfEDrGIeRG-tQwVGigW5exiQpHuvp5jcLFatHupEgWdPLbGKdST6zLmvlON8hhv3_KxgcvgBF1lp2PZ5vb2O4bALFtFXSNt1-y1zVVb80s01mEyXN99L8QHX5nH25H10jaP1JJuDYveUorVzRVEZlnIoRm6fTCFvdG_p5OWb4vilr2q1DeLuinZNkyLXa48Kk5FlkQOslGjMRo047KmOiRpLaQSD3smtF5tDWwcnAD4YoMWAAPuwU4mwkbf-AAH0AvoHAAD0PALbscRn-wIAAAAABFla):
```python
def test_ad_to_evol_map():
    triv_ops = ("S.S", "S.g", "g.S", "g.g", "V.V", "V3.V3", "T3.T3", "V8.V8", "T8.T8")
    # nf=3
    assert sorted(triv_ops) == get_ad_to_evol_map(3)
    # nf=3 + IC
    assert sorted(
        [*triv_ops, "T15.c+", "V15.c-"] #, "c+.c+"", "c-.c-"]
    ) == get_ad_to_evol_map(3, [4])
    # nf=3 + IC + IB
    assert sorted(
        [*triv_ops,"T15.c+", "V15.c-","b+.b+", "b-.b-"]
    ) == get_ad_to_evol_map(3, [4, 5])
    # nf=4 + IC + IB
    ks = sorted(
        [*triv_ops, "V15.V15", "T15.T15","T24.b+", "V24.b-"] # , "b+.b+", "b-.b-"]
    )
    assert ks == get_ad_to_evol_map(4, [4, 5])
    # nf=4 + IB
    assert ks == get_ad_to_evol_map(4, [5])
    # nf=6
    assert sorted(
        [*triv_ops, "T15.T15", "V15.V15", "T24.T24", "V24.V24", "T35.T35", "V35.V35"]
    ) == get_ad_to_evol_map(6)
```

also note that the diff snippet is *not* related to the comment, that I removed

**Comment by ichard26** [comment]
> it is correct to open a new issue, right? (instead of commenting

In hindsight it probably would've been better to explicitly allow follow up comments if the formatting was still unstable, but yes it's usually better still. I can't rename the issues, so I didn't want people to follow up with comments of how the output needs tweaking or there's now an invalid code produced bug, I can't edit the issue enough to turn into those other types.

By the way, are you sure this fails on master, black.now.sh's master version seems to be getting stuck on commit cac182 which doesn't have the instability fix. Here's what's happening locally:

```shell
~/programming/oss/black on  main [$?⇡] via  v3.8.5 (black) 
❯ black --version
black, version 21.4b0

~/programming/oss/black on  main [$?⇡] via  v3.8.5 (black) took 336ms 
❯ git log -n5 --oneline
67d5532 (HEAD -> main, tag: 21.4b0, upstream/master) Update CHANGES.md for release (#2129)
de7187a Issue 1629 has been closed, let's celebrate! (#2127)
eee949e Docker CI: Add missed Checkout step (#2128)
e5490e9 (master) Add Docker Github Action (#2086)
8672af3 Work around stability errors due to optional trailing commas (#2126)

~/programming/oss/black on  main [$?⇡] via  v3.8.5 (black) 
❯ vim test.py

~/programming/oss/black on  main [$?⇡] via  v3.8.5 (black) took 4s954ms 
❯ cat test.py
def test_ad_to_evol_map():
    triv_ops = ("S.S", "S.g", "g.S", "g.g", "V.V", "V3.V3", "T3.T3", "V8.V8", "T8.T8")
    # nf=3
    assert sorted(triv_ops) == get_ad_to_evol_map(3)
    # nf=3 + IC
    assert sorted(
        [*triv_ops, "T15.c+", "V15.c-"] #, "c+.c+"", "c-.c-"]
    ) == get_ad_to_evol_map(3, [4])
    # nf=3 + IC + IB
    assert sorted(
        [*triv_ops,"T15.c+", "V15.c-","b+.b+", "b-.b-"]#, "c+.c+"", "c-.c-", "b+.b+", "b-.b-"]
    ) == get_ad_to_evol_map(3, [4, 5])
    # nf=4 + IC + IB
    ks = sorted(
        [*triv_ops, "V15.V15", "T15.T15","T24.b+", "V24.b-"] # , "b+.b+", "b-.b-"]
    )
    assert ks == get_ad_to_evol_map(4, [4, 5])
    # nf=4 + IB
    assert ks == get_ad_to_evol_map(4, [5])
    # nf=6
    assert sorted(
        [*triv_ops, "T15.T15", "V15.V15", "T24.T24", "V24.V24", "T35.T35", "V35.V35"]
    ) == get_ad_to_evol_map(6)

~/programming/oss/black on  main [$?⇡] via  v3.8.5 (black) 
❯ black test.py --config=tests/empty.toml --check --safe 
would reformat test.py
Oh no! 💥 💔 💥
1 file would be reformatted.

~/programming/oss/black on  main [$?⇡] via  v3.8.5 (black) took 432ms 
1❯ black test.py --config=tests/empty.toml --check --safe --diff
--- test.py	2021-04-26 11:36:38.934699 +0000
+++ test.py	2021-04-26 11:37:31.408594 +0000
@@ -2,19 +2,28 @@
     triv_ops = ("S.S", "S.g", "g.S", "g.g", "V.V", "V3.V3", "T3.T3", "V8.V8", "T8.T8")
     # nf=3
     assert sorted(triv_ops) == get_ad_to_evol_map(3)
     # nf=3 + IC
     assert sorted(
-        [*triv_ops, "T15.c+", "V15.c-"] #, "c+.c+"", "c-.c-"]
+        [*triv_ops, "T15.c+", "V15.c-"]  # , "c+.c+"", "c-.c-"]
     ) == get_ad_to_evol_map(3, [4])
     # nf=3 + IC + IB
-    assert sorted(
-        [*triv_ops,"T15.c+", "V15.c-","b+.b+", "b-.b-"]#, "c+.c+"", "c-.c-", "b+.b+", "b-.b-"]
-    ) == get_ad_to_evol_map(3, [4, 5])
+    assert (
+        sorted(
+            [
+                *triv_ops,
+                "T15.c+",
+                "V15.c-",
+                "b+.b+",
+                "b-.b-",
+            ]  # , "c+.c+"", "c-.c-", "b+.b+", "b-.b-"]
+        )
+        == get_ad_to_evol_map(3, [4, 5])
+    )
     # nf=4 + IC + IB
     ks = sorted(
-        [*triv_ops, "V15.V15", "T15.T15","T24.b+", "V24.b-"] # , "b+.b+", "b-.b-"]
+        [*triv_ops, "V15.V15", "T15.T15", "T24.b+", "V24.b-"]  # , "b+.b+", "b-.b-"]
     )
     assert ks == get_ad_to_evol_map(4, [4, 5])
     # nf=4 + IB
     assert ks == get_ad_to_evol_map(4, [5])
     # nf=6
would reformat test.py
Oh no! 💥 💔 💥
1 file would be reformatted.
```

**Comment by felixhekhorn** [comment]
Yes, you're right - the snippet does work locally with `21.4b0` correctly

Just for my future education:
1. check the version in the playground explicitly (which is already printed, I realized now) and compare to `git`
2. continue on the issue if it is still not working 

do you agree? Maybe concerning 1. you may want to adjust your bug template? Or adjust the playground to display a warning, if not even with master?
